packages/material-ui/src/ButtonBase/Ripple.js

```diff
@@ -1,13 +1,27 @@
 import React from 'react';
 import PropTypes from 'prop-types';
 import clsx from 'clsx';
+import { makeStyles } from '@material-ui/styles';
 import { Transition } from 'react-transition-group';
 
+const useStyles = makeStyles(
+  {
+    root: {
+      width: props => props.rippleSize,
+      height: props => props.rippleSize,
+      top: props => -(props.rippleSize / 2) + props.rippleY,
+      left: props => -(props.rippleSize / 2) + props.rippleX,
+    },
+  },
+  { index: 100 },
```

Comment:
this is not an option. We can't hard code the injection order.

Reply:
I had a feeling this might be an issue.  The 100 is indeed arbitrary.  Would you consider a constant that could be used here that would represent any sheet that is intended to replace an inline style?  If not, I think I will have to go back and figure out how to refactor TouchRipple to make this work.

Reply:
We shouldn't override it, anywhere. Why is this even needed?

Reply:
@oliviertassinari, you are right, done.  This is my first time using the new @material/styles API.  Originally wrote this I was passing the entire props to useStyles, which was causing the styles from classes.root to get pulled in and kept this from working.  I didn't figure out my mistake there until after going the route of setting the index.  This is much cleaner now.